lib/daemon.py

```diff
@@ -160,12 +161,16 @@ def check_sh_is_running(pidfile):
         try:
             fd = os.open(pidfile, os.O_RDONLY)
             fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
-            os.close(fd)
             # pidfile not locked, so sh is terminated
         except OSError as e:
-            # pidfile is locked, so sh is running
-            isRunning = True
-            pass
+            if (e.errno == errno.EWOULDBLOCK):
+                # pidfile is locked, so sh is running
+                isRunning = True
+            else:
+                print("Error while testing lock in pidfile: %d (%s)" % (e.errno, e.strerror) , file=sys.stderr)
```

Comment:
I'm not sure what to assume in this case. When there is another error, we don't really know whether SH is running or not.

Reply:
Maybe we could exit with an error and print out a hint to the user to manually inspect the problem. Solutions for the user could be:
* manually remove the PID file
* manually kill the running process
* check permissions

Reply:
I'm not really sure what hint to give in this case.
We could just print the name of the pidfile, so the users knows which path to check.

If you want, i can include the three possible solutions you named in the error text, but suggesting solutions feels wrong as we actually have no idea what happened.

Reply:
Hopefully the error message is help enough. In case of error (which should usually not happen often) you should know what to do (especially when reading the error message errno and strerror).

So the current solution is OK.